st2actions/st2actions/runners/ssh/parallel_ssh.py

```diff
@@ -319,12 +320,25 @@ def _get_host_port_info(self, host_str):
         return (hostname, port)
 
     @staticmethod
-    def _generate_error_result(error_msg, tb):
+    def _generate_error_result(ex, error_msg, tb):
+        """
+        :param ex: Raised exception.
+        """
+        if isinstance(ex, SSHCommandTimeoutError):
+            return_code = -9
+        else:
+            return_code = 255
+
+        stdout = getattr(ex, 'stdout', None) or ''
+        stderr = getattr(ex, 'stderr', None) or ''
+
         error_dict = {
-            'error': error_msg,
-            'traceback': ''.join(traceback.format_tb(tb, 20)) if tb else '',
             'failed': True,
             'succeeded': False,
-            'return_code': 255
+            'return_code': return_code,
+            'stdout': stdout,
```

Comment:
Actually, I changed my mind :P

It would be weird if we only included one, but not the other (e.g. there is no stderr), so it's probably better to just always include it.